maintenance/cleanup.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# -----------------------------------------------------------------------------
+#   Copyright (C) 2017 University of Dundee. All rights reserved.
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation; either version 2 of the License, or
+#   (at your option) any later version.
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program; if not, write to the Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# ------------------------------------------------------------------------------
+
+
+# This Python script
+# does delete all the ratings and rois on all the images of user-1
+# through user-40 and of trainer-1 in dataset "Condensation"
+# The script also counts the number of the images in each
+# "Condensation" dataset and
+# warns if there is any other number than 32.
+# Further the script unlinks all tags from the images
+# in each "Condensation" dataset other than the tag
+# with id 4236
```

Comment:
Replace this commentary with



This script first finds and lists, then cleans up objects (ROIs, Ratings and Tags) attached to all images contained in a Dataset with a specified name ("Condensation") belonging to users user-1 through user-40 and for "trainer-1". The listing is done by each user on their own data. The information is collected and the final cleanup is done by administrator ("trainer-1"), which makes sure that also objects linked to images by other users will get unlinked or deleted.
The cleanup of ROIs and Ratings means deletion of these.
The cleanup of Tags just deletes the link between the Tag and the image, thus removing the Tag, but not deleting it.